lamindb/dev/db/_add.py

```diff
@@ -38,22 +48,34 @@ def add(  # type: ignore  # no support of different naming of args across overlo
     Example:
 
     >>> # insert a new record
-    >>> db.add(wetlab.experiment(name="My test", biometa_id=test_id))
+    >>> ln.add(wetlab.Experiment(name="My test", biometa_id=test_id))
     >>> # update an existing record
-    >>> experiment = ln.select(wetlab.experiment, id=experiment_id).one()
+    >>> experiment = ln.select(wetlab.Experiment, id=experiment_id).one()
     >>> experiment.name = "New name"
-    >>> db.add(experiment)
+    >>> ln.add(experiment)
+    >>> # check existence before inserting a new record
+    >>> ln.add(wetlab.Experiment, name="My test", biometa_id=test_id)
 
     Args:
         record: One or multiple records as instances of `SQLModel`.
+        use_fsspec: Whether to use fsspec.
     """
     if isinstance(record, list):
         records = record
-    else:
+    elif isinstance(record, sqm.SQLModel):
         records = [record]
+    else:
+        model = record
+        results = select(model, **fields).all()
```

Comment:
Could we do `.one()`, here?

Reply:
For this reason: If think you can only "add by field" if you have one unique record in mind.

Reply:
ok!

Reply:
This is why our pattern was: 
```
result = select().one_or_none()
if result is None:
    # add
```

Reply:
Yes, but I thought it doesn’t hurt to return all results? [](edited-with-Axolo)

Reply:
If the latter is relevant, maybe that's great!

But the downside is that one has an unexpected return type.

Reply:
Yes, that’s what I did [](edited-with-Axolo)

Reply:
So, one can't use it as an assert statement, but will always need to check that it's in fact just one record that's returned.

Reply:
Hence, I'd rather throw an error than returning a list if there a more return results in the query.

Reply:
but I think the point is to check whether any records with the field exists [](edited-with-Axolo)

Reply:
without caring much about the uniqueness [](edited-with-Axolo)

Reply:
With these fundamental additions to the API, I think the more minimalistic and expectable the behavior, the better.

Reply:
I'm not sure; merging now, we can discuss!